service/pkg/config/config.go

```diff
@@ -190,6 +191,22 @@ func (c *Config) Reload(ctx context.Context) error {
 		assigned = make(map[string]struct{})
 		orderedViper := viper.NewWithOptions(viper.WithLogger(slog.Default()))
 
+		// If an environment loader is present, configure orderedViper to read env directly.
+		for _, loader := range c.loaders {
+			if loader.Name() == LoaderNameEnvironmentValue {
```

Comment:
The env loader should already do this. And it should be configured by the consumer of the library.